docs/docs/gatsby-magic.md

```diff
@@ -46,16 +46,11 @@ Here is where you can find more information on [Routing in Gatsby](/docs/routing
 
 ## Redux
 
-Redux is a state management library often used together with React apps. Gatsby is also a React application and uses Redux behind-the-scenes to manage the state of your site. You don’t have to do anything with Redux directly in Gatsby for it to work.
+When you implement a Gatsby [Node API](https://www.gatsbyjs.org/docs/node-apis/), you are passed a collection of [actions](/docs/actions/) (equivalent to actions bound with [bindActionCreators in Redux](https://redux.js.org/api/bindactioncreators/)). When using the supplied actions, you are manipulating state that is created, and relied upon, by Gatsby itself such as created pages and webpack config.
 
-Redux helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test; that’s why Gatsby uses it as a core technology under the hood. On top of that, Redux provides a great developer experience, such as [live code editing combined with a time-traveling debugger](https://github.com/reduxjs/redux-devtools). Redux is bundled with Gatsby by default to provide automatic state management, and it’s also available as a standalone package on npm.
+Once the build is complete, the work of Redux is done and it is not part of the bundle delivered to the browser unless required independently by the application itself.
 
-In order to use Redux for custom state management in a Gatsby site, you'll need to hook into two of Gatsby's extension points:
-
-- Wrap the root element in your Gatsby markup once using `wrapRootElement`, an API supporting both Gatsby’s server rendering and browser JavaScript processes.
-- Create a custom Redux store with reducers and initial state, providing your own state management functionality outside of Gatsby.
-
-Check out the Using Redux example with [./gatsby-ssr.js](https://github.com/gatsbyjs/gatsby/blob/master/examples/using-redux/gatsby-ssr.js) and [./gatsby-browser.js](https://github.com/gatsbyjs/gatsby/blob/master/examples/using-redux/gatsby-browser.js) files to see how this is implemented. You can also check out more information on the official [Redux](https://redux.js.org/introduction/getting-started) docs.
+If using a data store such as Redux in your Gatsby application, the [using-redux](https://github.com/gatsbyjs/gatsby/blob/master/examples/using-redux) example is a good starting point for connecting it to the pages of your Gatsby application.
```

Comment:
The example is helpful, but is there a reason to remove the breakdown the previously existed here? In my estimation that was all useful info.

Reply:
Absolutely agree, I mainly took it out to address the issue (#20565) saying it's out of scope for this page, but it's really useful info to have.

Just pushed up an update that drops it on a new part of the https://www.gatsbyjs.org/docs/adding-app-and-website-functionality/ reference guide. What do you think?

Couple of notes about possible further work:
- Could also make it a more generic section that talks more generally about adding a data store.
- Might be an even more comfortable home and more findable to have it structured out into a recipe.

Reply:
The intent of the Gatsby Magic doc is to clear up any confusion about how Gatsby works, and people might be confused about Redux (as we all were, apparently). It's also a conceptual guide, though I recognize that it could stray pretty far into Internals. A level-setting introduction is still useful in my opinion...it just needs to be tweaked for accuracy. Something like:

> Redux is a state management library often used together with React apps. Gatsby is also a React application and uses Redux internally for managing state and sending out events. Unless you want to handle extra state management yourself using Redux, you wouldn't need to interface with it directly in Gatsby for it to work.

Reply:
Agreed. I think @wardpeet's intro approach of not mentioning React invites confusion a little less as it really doesn't come into contact with Gatsby's Redux at all, so I've brought that one in.